dali/operators/image/distortion/jpeg_compression_distortion_op_gpu.cu

```diff
@@ -38,17 +37,63 @@ class JpegCompressionDistortionGPU : public JpegCompressionDistortion<GPUBackend
   std::vector<int> quality_;
 };
 
+template <int ndim>
+TensorListShape<ndim - 1> unfold_dim(const TensorListShape<ndim> &shapes) {
+  static_assert(ndim > 1,
+                "Can't reduce dimentionality of dynamic, or single dimentional TensorListShape");
+
+  constexpr static int out_dim = ndim - 1;
+
+  using OutShapeType = TensorShape<out_dim>;
+  TensorListShape<out_dim> result;
```

Comment:
maybe we can go through the original shape and preallocate the result to the expected number of elements
```
int out_nsamples = 0;
for (int i = 0; i < shapes.size(); ++i) {
  out_nsamples += shapes[i][0];
}
result.reserve(out_nsamples);
```
or `resize` (in such case we won't use append/push_back below)

Reply:
Done